.lgtm.yml

```diff
@@ -12,6 +12,7 @@ extraction:
     python_setup:
       requirements_files: false
       requirements:
+        - contourpy
```

Comment:
It is an important question. I suppose we'd have to consider the number of dependencies that are added and their potential to cause problems. Pure python dependencies are relatively safe as they are platform agnostic, those with compiled C extensions are more complicated (need platform-specific wheels and C runtime libraries) and those with compiled C++ extensions yet more complicated (need C++ runtime as well as C).

ContourPy only requires NumPy and a C++ runtime, as is available for all the platforms supported by NumPy.

I'll do an experiment to look at the dependencies:

1. Create an empty conda environment (Python 3.10 on Linux x86_64). This straightaway gives you C (`libgcc-ng` and `libgomp`) and C++ (`libstdcxx-ng`) runtimes, version 11.2.0. Names and versions of the runtimes will differ on different platforms.
2. `conda install -c conda-forge bokeh` adds 28 packages, 19 of which are compiled C (mostly `numpy` and image related) and none are compiled C++.
3. `conda install -c conda-forge contourpy` installs just one new package (`contourpy`) and updates the C and C++ runtimes to 12.1.0. This is the first compiled C++ library added. The runtime is only updated because it is a recent release, 3 weeks ago.
4. `conda install -c conda-forge pandas` installs 7 new packages, 2 of which include at least one compiled C++ library (`numexpr`, `pandas`) and 1 is compiled C (`bottleneck`).
5. Geopandas is split into `geopandas-base` and `geopandas` on conda but not PyPI (they recommend using conda because with pip it "can be a challenge to install"). `conda install -c conda-forge geopandas-base` installs 13 new packages, 8 are compiled C and 3 compiled C++ (`geos`, `proj`, `shapely`). This is where I start to get nervous.
6. `conda install -c conda-forge geopandas` (after `geopandas-base` installed) installs 102 packages. That is not a typo. I haven't counted the C/C++ libraries here, but some noteworthy packages are `fiona`, `gdal`, `matplotlib-base`, `scikit-learn`.

I would include `contourpy` as a compulsory dependency, and I would be happy to include `pandas` too. In the event of any problem with either I think/know we can guarantee a fast response. But I would defer to the opinions of others here as I have a vested interest. I would not support adding any form of `geopandas` as a compulsory dependency.

Reply:
FWIW I should also say that `contourpy` will be a compulsory dependency of `matplotlib` 3.6.0.

Reply:
> I would include contourpy as a compulsory dependency, and I would be happy to include pandas too. In the event of any problem with either I think/know we can guarantee a fast response. But I would defer to the opinions of others here as I have a vested interest. I would not support adding any form of geopandas as a compulsory dependency.

I concur with this.

Reply:
I will leave adding Pandas as a hard dependency for another PR.